pkg/collector/edp/collector.go

```diff
@@ -2,69 +2,204 @@ package edp
 
 import (
 	"context"
-	"maps"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
 	"time"
 
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/trace"
-	"go.uber.org/zap"
 
 	"github.com/kyma-project/kyma-metrics-collector/pkg/collector"
 	"github.com/kyma-project/kyma-metrics-collector/pkg/resource"
 	"github.com/kyma-project/kyma-metrics-collector/pkg/runtime"
-	skrcommons "github.com/kyma-project/kyma-metrics-collector/pkg/skr/commons"
 )
 
 type Collector struct {
-	scanners []resource.Scanner
-	logger   *zap.Logger
+	EDPClient *Client
+	scanners  []resource.Scanner
 }
 
 var _ collector.CollectorSender = &Collector{}
 
-func NewCollector(scanner ...resource.Scanner) collector.CollectorSender {
+func NewCollector(EDPClient *Client, scanner ...resource.Scanner) collector.CollectorSender {
 	return &Collector{
-		scanners: scanner,
+		EDPClient: EDPClient,
+		scanners:  scanner,
 	}
 }
 
 func (c *Collector) CollectAndSend(ctx context.Context, runtime *runtime.Info, previousScans collector.ScanMap) (collector.ScanMap, error) {
-	childCtx, span := otel.Tracer("").Start(ctx, "collect",
+	var errs []error
+
+	childCtx, span := otel.Tracer("").Start(ctx, "kmc.collect_scans_and_send_measurements",
 		trace.WithAttributes(
+			attribute.String("instance_id", runtime.InstanceID),
+			attribute.String("runtime_id", runtime.RuntimeID),
+			attribute.String("sub_account_id", runtime.SubAccountID),
+			attribute.String("global_account_id", runtime.GlobalAccountID),
+			attribute.String("shoot_name", runtime.ShootName),
 			attribute.String("provider", runtime.ProviderType),
```

Comment:
lets make these attribute names constants

Reply:
They are constants now in the `otel` package